xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp

```diff
@@ -1177,13 +1177,19 @@ DemuxPacket* CDVDDemuxFFmpeg::Read()
           if (pPacket->dts != DVD_NOPTS_VALUE &&
               (pPacket->dts > m_currentPts || m_currentPts == DVD_NOPTS_VALUE))
             m_currentPts = pPacket->dts;
+          else if (pPacket->pts != DVD_NOPTS_VALUE &&
+              (pPacket->pts > m_currentPts || m_currentPts == DVD_NOPTS_VALUE))
+            m_currentPts = pPacket->pts;
```

Comment:
That should be dts. Pts is not monotonic at this stage

Reply:
Without this we discard video I frames, which causes large stalls.
How about we only update from pts when `m_currentPts == DVD_NOPTS_VALUE`?

```
          else if (pPacket->pts != DVD_NOPTS_VALUE && m_currentPts == DVD_NOPTS_VALUE))
            m_currentPts = pPacket->pts;
```

Reply:
Sorry, only had a a smartphone yesterday. You added pts as fallback if dts is not set. This is perfectly fine. Sorry for the noise.